includes/REST_API/Stories_Controller.php

```diff
@@ -157,26 +157,35 @@ public function get_item_schema() {
 	}
 
 	/**
-	 * Filters the orderby query to filter first all the current user's posts and then the rest.
+	 * Filters query clauses to sort posts by the author's display name.
 	 *
-	 * @param string    $orderby Original orderby clause.
-	 * @param \WP_Query $query WP_Query object.
-	 * @return string Orderby clause.
+	 * @param string[] $clauses Associative array of the clauses for the query.
+	 * @param WP_Query $query   The WP_Query instance.
+	 *
+	 * @return array Filtered query clauses.
 	 */
-	public function filter_posts_orderby( $orderby, $query ) {
+	public function filter_posts_clauses( $clauses, $query ) {
 		global $wpdb;
+
 		if ( Story_Post_Type::POST_TYPE_SLUG !== $query->get( 'post_type' ) ) {
-			return $orderby;
+			return $clauses;
 		}
 		if ( 'story_author' !== $query->get( 'orderby' ) ) {
-			return $orderby;
+			return $clauses;
 		}
 
-		$current_user = get_current_user_id();
-		if ( ! $current_user ) {
-			return $orderby;
+		// Usage of users/usermeta tables is highly discouraged in VIP context.
```

Comment:
If this not supported in VIP GO, then this option should be hidden in the UI. @BrittanyIRL Thoughts?

Reply:
IMHO that's something for a separate issue to discuss, especially since  dashboard team lacks context here.

Why?

First of all, WP.com is currently far away and not really relevant. For now, this check is basically a failsafe to prevent breakage for the very unlikely event of someone trying this plugin on VIP.

Second, I assume that on WP.com / VIP there are other ways to achieve this kind of functionality. So once that platform becomes relevant, we can act as needed.

Third, we should not be disabling such a small UI  thing based on environment. We should much rather provide a fallback or alternative solution to achieve the same result.

Reply:
Opened #2413 to track WP.com compatibility